sys/cheri/revoke.h

```diff
@@ -133,8 +137,10 @@ cheri_revoke_is_revoked(const void * __capability cap)
 static const size_t VM_CHERI_REVOKE_GSZ_MEM_NOMAP = sizeof(void * __capability);
```

Comment:
This probably belongs under the #ifdef, too.  While it will compile for !CHERI, it will not be a sensible value.  Once you do that you can basically just remove the #endif up on line 89 and let it go all the way down to your new #endif currently on line 143.  Arguably you could widen it even further so you just expose the flags at the bottom of the file.